app/src/main/java/net/squanchy/contest/ContestService.java

```diff
@@ -0,0 +1,39 @@
+package net.squanchy.contest;
+
+import net.squanchy.remoteconfig.RemoteConfig;
+import net.squanchy.service.firebase.FirebaseAuthService;
+import net.squanchy.service.firebase.FirebaseDbService;
+import net.squanchy.service.firebase.model.FirebaseAchievements;
+
+import io.reactivex.Single;
+import io.reactivex.functions.BiFunction;
+import io.reactivex.schedulers.Schedulers;
+
+public class ContestService {
+
+    private final RemoteConfig remoteConfig;
+    private final FirebaseDbService dbService;
+    private final FirebaseAuthService authService;
+
+    public ContestService(RemoteConfig remoteConfig, FirebaseDbService dbService, FirebaseAuthService authService) {
+        this.remoteConfig = remoteConfig;
+        this.dbService = dbService;
+        this.authService = authService;
+    }
+
+    public Single<ContestStandings> standings() {
+        return authService.ifUserSignedInThenObservableFrom(dbService::achievements)
+                .first(FirebaseAchievements.empty())
```

Comment:
It is to maybe re-use it, I actually did in the intervention for [https://github.com/rock3r/squanchy/pull/171#discussion_r108206048](url).
If I just access the contest standings, I don't try to add an achievement. Still trying to figure out a more reactive way to do it